File: 1331/solution.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        int rank = 1;
        int[] nums = arr.clone();
        Map<Integer, Integer> map = new HashMap<>();
        Arrays.sort(nums);
        for(int i: nums){
            if(!map.containsKey(i))
                map.put(i, rank++);
        }
        for(int i=0;i<arr.length;i++){
            arr[i] = map.get(arr[i]);
        }
        return arr;         
    }
}

File: 1331/solution.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        rank = 1 
        nums = sorted(arr)
        rankMap = {}
        for i in nums:
            if i not in rankMap:
                rankMap[i] = rank
                rank+=1 
        for i in range(len(arr)):
            arr[i] = rankMap[arr[i]]
        return arr
